src/pint/models/astrometry.py

```diff
@@ -903,25 +904,73 @@ def as_ECL(self, epoch=None, ecl="IERS2010"):
         epoch : `astropy.time.Time` or Float, optional
             new epoch for position.  If Float, MJD(TDB) is assumed.
             Note that uncertainties are not adjusted.
+        ecl : str, optional
+            Obliquity for PulsarEcliptic frame
 
         Returns
         -------
         pint.models.astrometry.AstrometryEcliptic
-
-        Raises
-        ------
-        ValueError
-            If obliquity is already set and a different obliquity is requested
         """
-        if not (ecl == self.ECL.value):
-            raise ValueError(
-                "Cannot convert from obliquity '%s' to requested obliquity '%s'"
-                % (self.ECL.value, ecl)
-            )
-        m = copy.deepcopy(self)
-        if epoch is not None:
-            m.change_posepoch(epoch)
-        return m
+
+        # change epoch only
+        if ecl == self.ECL.value:
+            print("here")
+            m = copy.deepcopy(self)
+            if epoch is not None:
+                m.change_posepoch(epoch)
```

Comment:
This should be covered in the change_posepoch tests. Don't we have tests for it?

Reply:
Not sure entirely what you mean.  This is just so that you don't get an error for a transformation (say) from IERS2010 to IERS2010.

Reply:
In the changed files, there is a codecov warning saying "Added line #L920 was not covered by tests".